### DISCUSSION TITLE
Update/delete if condition is met

### DISCUSSION THREAD

**Discussion by liadmat** [prisma/prisma#6633] [Q&A] [answered]
Hi all,

Let's say I have an app for task lists, and I only want to let users update/delete tasks if and only if they're the task owner.

My schema looks something like this:

```prisma
model User {
  username       String  @id
  tasks          Task[]
}

model Task {
  id             String  @id @default(uuid())
  ownerUsername  String
  owner          User    @relation(fields: [ownerUsername], references: [username])
}
```
What I'm looking for is something like:

```
prisma.task.delete({ where: { id, ownerUsername: currentUsername } })
```
However I'm getting this error:
```
Type '{ id: string; ownerUsername: string; }' is not assignable to type 'TaskWhereUniqueInput'.
  Object literal may only specify known properties, and 'ownerUsername' does not exist in type 'TaskWhereUniqueInput'.
```
I can split this to 2 calls, `findUnique` + `delete`, but this means 2 roundtrips to the DB. Perhaps I can do this in a transaction, but it seems kind of overkill.

What's the recommended approach for this?

Thanks!

**Comment by janpio** [comment]
You need to use `deleteMany`, as you are not deleting by a unique field but by one that could in theory remove multiple entries.

**Reply by liadmat** [reply]
Thanks! This will indeed work, however I have a couple of questions:

1. Since I'm using a specific unique item ID (in _addition_ to `ownerUsername`), I feel like `deleteMany` is a bit misleading. Is there no way to achieve this with `delete`? After all the item _is_ unique.
2. Is there a way to retrieve the deleted item, such as when using `delete`?

**Reply by janpio** [reply]
Hm you are right, I might have misread your initial question a bit.

You could try to add a `@@unique[id,ownerUsername]` to the model as that will let Prisma know that this combination is unique as well. (Feels a bit the wrong way though...)

**Reply by liadmat** [reply]
I've considered this but it feels a little hacky, because the `id` is globally unique regardless of `ownerUsername`, and this approach will break once the authorization rule is a little more complex (e.g. multiple owners).

**Reply by liadmat** [reply]
@janpio , is there any approach Prisma recommends here? Seems like a very common usecase.

**Reply by janpio** [reply, marked as answer]
Yes, I think I mentioned them both: Use `deleteMany` if you are not deleting by a unique field/index, or `delete` with an index that actually defines the unique columns so our Client can understand that. Those are the two options to delete things in Prisma. Or is there a reason why you can not include an unique index that matches your data?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
